.circleci/config.yml

```diff
@@ -712,6 +712,12 @@ workflows:
           iOS: "14.5"
           device: "iPhone 12 Pro Max"
           context: Slack Orb
+      - spm-test-job:
+          name: "swift test; Xcode 13.2.1; iOS 15.2"
+          xcode: "13.2.1"
+          iOS: "15.2"
+          device: "iPhone 12 Pro Max"
```

Comment:
Several snapshot (screenshot) tests fail on [this build](https://app.circleci.com/pipelines/github/mapbox/mapbox-navigation-ios/7220/workflows/62c573b1-b70a-4460-adc1-cd635b91964a/jobs/60206):

```
		InstructionsBannerViewSnapshotTests.testAbbreviateInstructions()
		InstructionsBannerViewSnapshotTests.testAbbreviateInstructionsIncludingDelimiter()
		InstructionsBannerViewSnapshotTests.testAbbreviateWestFremontAvenue()
		InstructionsBannerViewSnapshotTests.testExitShields()
		InstructionsBannerViewSnapshotTests.testGenericShieldAndExitViewWithCustomDayStyle()
		InstructionsBannerViewSnapshotTests.testGenericShields()
		InstructionsBannerViewSnapshotTests.testLongDistance()
		InstructionsBannerViewSnapshotTests.testMultilinePrimary()
		InstructionsBannerViewSnapshotTests.testPrimaryShieldAndSecondary()
		InstructionsBannerViewSnapshotTests.testSinglelinePrimaryAndSecondary()
		InstructionsBannerViewSnapshotTests.testSweEngLongDistance()
		InstructionsBannerViewSnapshotTests.testUkrainianLongDistance()
```

Font metrics may have changed subtly between minor versions of iOS. You can either SSH into the build machine or run the tests locally to compare the screenshots visually. If the output is acceptable, we can check in the new screenshots for iOS 15.2 specifically.

Reply:
> If the output is acceptable, we can check in the new screenshots for iOS 15.2 specifically.

I.e. we can have separate screenshots for different SDK versions? How can we do that?

Reply:
Oh, it looks like we’ve only retained the ability to vary it by iPhone model rather than iOS version. That probably changed for v2.0.0 when we migrated to a different snapshot testing framework. The first step would be to diff the snapshots to see if they’re visually alike and if the difference is even caused by the OS upgrade versus some other kind of failure.

Reply:
Btw does this snapshotting framework generate diff images? I remember I used something like this in the past.

Reply:
@1ec5 I see 2 options here: either we simply skip those tests for iOS 15(i.e. for only this new particular job only) or we add one more dimension "OS version" [here](https://github.com/mapbox/mapbox-navigation-ios/blob/40440f039d5b6799ae0f2df66dcc66a1524b32d9/Tests/MapboxNavigationTests/Snapshot%2B%2B.swift#L27). WDYT?

Reply:
@1ec5 well, but not clear how to record snapshots in that case. Imagine you run snapshot tests in record mode on iOS 15 and what would we do? Just overwrite files in "default" folder? And then manually move updated files to dedicated folder?

> so that we don’t have to update the snapshot tests every time we upgrade CI to a minor version

Btw I remember from my past experience that such issues are usually caused only by major iOS version upgrade, should we distinguish only by major version then?